src/Strategy/Node/Npm/PackageLockV3.hs

```diff
@@ -0,0 +1,435 @@
+      --
+      -- "workspace-a": {
+      --   "name": "someWorkspacePkgName",
+      --   "version": "1.0.0",
+      --   "dependencies": {
+      --     "someWorkspaceDirectDep": "^1.0.0",
+      --   }
+      --  },
+      --
+      -- Now this, @someWorkspaceDirectDep, might be resolved at:
+      --    - top path: node_modules/someWorkspaceDirectDep
+      --    - vendor path: workspace-a/node_modules/someWorkspaceDirectDep
+      --
+      -- If the @someWorkspaceDirectDep was resolved with top path,
+      -- vendored path (within workspace-a) will not exist. For this reason,
+      -- always prefer vendored path, if such path does not exist fallback to
+      -- top level path.
+      --
+      -- Direct Dependencies within workspace pkg are treated as direct.
+      -- Direct Peer Dependencies within workspace pkg are treated as direct.
+      -- Development Dependencies within workspace pkg are treated as direct development.
+      PackageLockV3WorkSpace workspaceRootPath -> do
+        let directDeps :: [PackagePath]
+            directDeps =
+              (vendoredPathElseTopLevelPath workspaceRootPath)
+                <$> concatMap
+                  Map.keys
+                  [ plV3PkgDependencies pkgMetadata
+                  , plV3PkgDevDependencies pkgMetadata
+                  , plV3PkgPeerDependencies pkgMetadata
+                  ]
+
+        let resolvedDeps :: [Dependency]
+            resolvedDeps = mapMaybe toDependency directDeps
+
+        traverse_ direct resolvedDeps
+
+      -- For typical dependency packages, for instance:
+      --
+      -- "node_module/someDirectDep": {
+      --   "version": "1.2.3",
+      --   "resolved": "https://registry.npmjs.org/someDirectDep/-/someDirectDep-1.2.3.tgz",
+      --   "dependencies": {
+      --     "someTransitiveDepB": "^1.0.0",
+      --   }
+      --  },
+      --
+      -- Now this, @someTransitiveDepB, might be resolved at:
+      --    - top path: node_modules/someTransitiveDepB
+      --    - vendor path: node_modules/a/node_modules/someTransitiveDepB
+      --
+      -- If the @someTransitiveDepB was resolved with top path,
+      -- vendored path (within node_modules/a/) will not exist. For this reason,
+      -- always prefer vendored path, if such path does not exist fallback to
+      -- top level path.
+      --
+      -- Only 'dependencies' field is used to infer transitive dependencies.
+      --
+      -- For each transitive dependency, Adds edge between parent package, and transitive dep.
+      PackageLockV3PathKey prefixPath depPackageName -> do
+        let currentDep :: Maybe Dependency
+            currentDep = toDependency (PackageLockV3PathKey prefixPath depPackageName)
+
+        let vendorPathPrefix :: Text
+            vendorPathPrefix = prefixPath <> (unPackageName depPackageName)
+
+        let deepDeps :: [PackagePath]
+            deepDeps =
+              (vendoredPathElseTopLevelPath vendorPathPrefix)
+                <$> concatMap
```

Comment:
[required]: Use `map` instead of `<$>` or `fmap` for lists:
```suggestion
              map (vendoredPathElseTopLevelPath vendorPathPrefix) $
                concatMap
```